decidim-core/spec/controllers/active_storage/direct_uploads_controller_spec.rb

```diff
@@ -0,0 +1,166 @@
+          sign_in user
+        end
+
+        it "returns unauthorized" do
+          post(:create, params:)
+
+          expect(response).to have_http_status(:unauthorized)
+        end
+      end
+
+      context "when the upload is successful" do
+        before do
+          request.env["decidim.current_organization"] = organization
+          request.headers["HTTP_REFERER"] = "http://#{organization.host}"
+          sign_in user
+        end
+
+        it "returns success" do
+          post(:create, params:)
+
+          expect(response).to have_http_status(:ok)
+          expect(response.body).to include("content_type")
+        end
+      end
+
+      context "when the attachment is not allowed" do
+        before do
+          request.env["decidim.current_organization"] = organization
+          request.headers["HTTP_REFERER"] = "http://#{organization.host}"
+          sign_in user
+        end
+
+        context "when content_type is not allowed" do
+          let(:content_types) { %w(image/jpeg) }
+
+          it "returns renders unprocessable entity" do
+            post(:create, params:)
+
+            expect(response).to have_http_status(:unprocessable_entity)
+          end
+        end
+
+        context "when extension is not allowed" do
+          let(:extensions) { %w(jpeg) }
+
+          it "returns renders unprocessable entity" do
+            post(:create, params:)
+
+            expect(response).to have_http_status(:unprocessable_entity)
+          end
+        end
+
+        context "when size is not allowed" do
+          let(:file_size) { { "default" => 0, "avatar" => 0 } }
+
+          it "returns renders unprocessable entity" do
+            post(:create, params:)
+
+            expect(response).to have_http_status(:unprocessable_entity)
+          end
+        end
+
+        context "when extension not matching content_type" do
+          let(:blob) do
+            {
+              filename: "hello.pdf",
+              byte_size: 6,
+              checksum:,
+              content_type: "text/plain"
+            }
+          end
+
+          it "returns renders unprocessable entity" do
+            post(:create, params:)
+
+            expect(response).to have_http_status(:unprocessable_entity)
+          end
+        end
+
+        context "when extension and mimetype differs from the content file" do
+          let(:extensions) { %w(pdf) }
+          let(:content_types) { %w(application/pdf) }
+          let(:blob) do
+            {
+              filename: "odt_file_reported_as.pdf",
+              byte_size: 6,
+              checksum: Digest::MD5.base64digest("odt_file_reported_as.pdf"),
+              content_type: "application/pdf"
+            }
+          end
+
+          it "returns renders unprocessable entity" do
+            post(:create, params:)
+
+            expect(response).to have_http_status(:success)
+
+            pp ActiveStorage::Blob.last.inspect
```

Comment:
Is this a debugging statement or is intended?

Reply:
actually is a debug spec, where i tried to identify if a file that has a wrong content type reported can be prevented to be uploaded, but it seems that we can't, as the actual file is not uploaded at this time. That url just authorizes the upload to external provider. 
We have a spec in disk_controller, where we validate that a file that is being uploaded with wrong mime / extension is not being sent by the server with the actual content type ( helps us to identify exe files reported and uploaded as pdf or anything else )

Reply:
removed in [d7e0696](https://github.com/decidim/decidim/pull/11851/commits/d7e069684328bd709550f1a48e443b14573de718)